### ISSUE TITLE
[bug] Conan 2.3.2 detects incorrect version of mingw-w64-x86_64-gcc 14.1.0-3

### ISSUE THREAD

**Issue by jwatson0** [conan-io/conan#16352]
### Describe the bug

Summary:
Running `conan profile detect` on `MSYS2` version `3.5.3` with gcc package `mingw-w64-x86_64-gcc` version `14.1.0-3` detects `compiler.version=3` instead of `14`.  I think conan is getting confused by the `Rev3` in the `cc` version string.

Note that you shouldn't rely solely on detected profiles, but this seems like a bug.

```
# conan profile detect -v -f
detect_api: Found cc=gcc-3
detect_api: gcc C++ standard library: libstdc++11

Detected profile:
[settings]
arch=x86_64
build_type=Release
compiler=gcc
compiler.cppstd=gnu98
compiler.libcxx=libstdc++11
compiler.version=3
os=Windows
```

Other details:
```
# pacman -Q mingw-w64-x86_64-gcc -v
Root      : /
Conf File : /etc/pacman.conf
DB Path   : /var/lib/pacman/
Cache Dirs: /var/cache/pacman/pkg/
Hook Dirs : /usr/share/libalpm/hooks/  /etc/pacman.d/hooks/
Lock File : /var/lib/pacman/db.lck
Log File  : /var/log/pacman.log
GPG Dir   : /etc/pacman.d/gnupg/
Targets   : mingw-w64-x86_64-gcc

# which cc
/mingw64/bin/cc

# cc --version
cc.exe (Rev3, Built by MSYS2 project) 14.1.0
Copyright (C) 2024 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.

# conan --version
Conan version 2.3.2

# uname -a
MSYS_NT-10.0-17763 EC2AMAZ-XXXXXXX 3.5.3.x86_64 2024-05-06 06:45 UTC x86_64 Msys
```

### How to reproduce it

- Install MSYS2
- Install gcc version `14.1.0-3`
    `pacman -S mingw-w64-x86_64-gcc=14.1.0-3`
- Run `conan profile detect`
- Examine created default profile

**Comment by AbrilRBS** [comment]
Hi @jwatson0 thanks a lot for taking the time to report this issue. We can reproduce this (And it in fact is getting confused with the Rev3 number), and we'll try to target a fix for this  for 2.4 release.

The issue here comes from us failing to parse --version for gcc when we dont expect gcc to be called :)

**Comment by jwatson0** [comment]
Note this looks to have appeared in `2.3.0`.
Conan `2.2.3` detects this correctly.
```
# conan --version
Conan version 2.2.3

# conan profile detect -v -f
detect_api: Found gcc 14.1
detect_api: gcc>=5, using the major as version
detect_api: gcc C++ standard library: libstdc++11

Detected profile:
[settings]
arch=x86_64
build_type=Release
compiler=gcc
compiler.cppstd=gnu17
compiler.libcxx=libstdc++11
compiler.version=14
os=Windows

# cc --version
cc.exe (Rev3, Built by MSYS2 project) 14.1.0
```

**Comment by memsharded** [comment]
https://github.com/conan-io/conan/pull/16362 has been merged, it will be in next Conan 2.4 soon (hopefully tomorrow)
